spec/core/data_structures.md

```diff
@@ -236,6 +238,20 @@ type DuplicateVoteEvidence struct {
 }
 ```
 
+Valid Duplicate Vote Evidence must adhere to the following rules:
+
+- Validator Address, Height, Round and Type of vote must be the same for both votes
+
+- BlockID must be different for both votes (BlockID can be for a nil block)
+
+- Validator must have been in the validator set at that height
+
+- Vote signature must be valid (using the chainID)
+
+- Time must be equal to the block time
```

Comment:
Isn't the Timestamp in the votes the validator's local clock? Not sure there's any validation we can/should do on the Timestamp (and I don't see any in the current code)

Reply:
Nvm was looking at outdated version.